src/path.cpp

```diff
@@ -94,4 +123,15 @@ Writer& Path::serialize(Writer& writer, bool pad_after_length) const
   return writer;
 }
 
+void Path::print() const
+{
+  std::cout << "PATH: ";
+  for (unsigned int ii=0; ii< path_buf_.size(); ii++)
+  {
+    std::cout << std::hex << std::setfill('0') << std::setw(2) << (unsigned short int)path_buf_.at(ii);
+	if ( (ii+1)%2 == 0 )  std::cout << " ";
+  }
+  std::cout << std::dec << std::endl;
+}
```

Comment:
I know the pattern thus far in this code was to write to cout and avoid an explicit ros dependency, but I'd rather just pull in `ros/console.h` and use the established logging framework— then everything can be left on as debug-level outputs and simply enabled at runtime as required.

Note also that rather than printing here, you can implement a stream operator, which allows the printing code to decide on logging level rather than hard-coding it in the `print` function. Eg:

```
std::ostream& operator<<(std::ostream& os, const Path& p)
{
  os << "PATH ";
  // for ...
  return os;
}
```

And then later:

```
ROS_DEBUG_STREAM("Processing " << my_path);
```

Thoughts on this as an alternative approach?

Reply:
Good idea to overload the operator "<<"
Doing o, as far as I understand we keep the no-dependency on ros/console at this level of the library, but it can be used with ROS_xx_STREAM()
thanks.

Reply:
Another possibility if you want to log while avoiding an explicit `ros/control.h` include is to use `console_bridge`:

http://wiki.ros.org/console_bridge